### PR TITLE
Refactor room generation logic

### LINKED ISSUE #65: Allow Rooms to be marked as `unique` and/or `required`
Some rooms have to exist for the game to proceed. Some rooms should be unique. When a floor is generated, both of these traits should be consulted, i.e. a floor should contain at most one of each unique room and at least one of each required room.

### PR DESCRIPTION
**Pull request by xmunoz**
- Remove unused get_room logic
- Eliminate *args where not needed
- Use a static list instead of a generator for room selection
- Add REQUIRED class attribute to rooms
- Add tests for new logic

### CODE REVIEW

File: tests/unit/cronenbroguelike/test_rooms.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from cronenbroguelike import rooms
+from unittest.mock import patch
+import pytest
+
+# TODO: introduce some room test fixtures
+
+@patch("cronenbroguelike.rooms._rooms_for_theme")
```
Comment (flosincapite) [inline tests/unit/cronenbroguelike/test_rooms.py:7]: privacy :-1:

**Review by flosincapite — APPROVED** [review]

**Comment by flosincapite** [comment]
YOU FORCE MERGED

WITH BLACK COMPLAINING

TESTS ARE FAILING AT HEAD

**Comment by flosincapite** [comment]
RED MOTHERFUCKING ALERT

WHERE IS MY DEV OPS PERSON

OH WAIT IT WAS MY DEV OPS PERSON WHO RAW DOGGED THE REPO WITHOUT A CONDOM NOOOOO

File: cronenbroguelike/rooms.py [hunk 2]
```diff
@@ -196,8 +193,10 @@ class _AltarRoom(_Room):
     )
     _THEMES = ["cathedral"]
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    REQUIRED = True
```
Comment (abjose) [inline cronenbroguelike/rooms.py:196]: Nice!

**Pull request merged**